lib/cocina/generator/schema_base.rb

```diff
@@ -84,19 +84,17 @@ def datatype_from_doc_type(doc)
       end
 
       def datatype_from_doc_names(doc)
-        if defined_datatypes?(doc)
-          doc.one_of.map(&:name).join(' | ')
-        elsif any_datatype?(doc)
-          'Types::Nominal::Any'
-        end
+        return unless defined_datatypes?(doc)
+
+        doc.one_of.map(&:name).join(' | ')
       end
 
       def defined_datatypes?(doc)
-        doc.one_of&.map(&:name).all? { |name| name.present? && schemas.include?(name) }
+        doc.one_of&.map(&:name)&.all? { |name| name.present? && schemas.include?(name) }
       end
 
       def any_datatype?(doc)
-        relaxed || doc.one_of&.map(&:type).all? { |o| %w[integer string].include?(o) }
+        relaxed || doc.one_of&.map(&:type)&.all? { |o| %w[integer string].include?(o) }
```

Comment:
With the above logic change, we need to be a little more careful with mapping over the oneOfs.